window/src/os/wayland/connection.rs

```diff
@@ -11,8 +11,7 @@ use mio::{Evented, Events, Poll, PollOpt, Ready, Token};
 use smithay_client_toolkit as toolkit;
 use std::cell::RefCell;
 use std::collections::HashMap;
-use std::io::Read;
-use std::os::unix::io::FromRawFd;
+use std::os::unix::{fs::FileExt, io::FromRawFd};
```

Comment:
nit: I prefer this style as it feels easier to read to me, even though it is slightly more verbose, it feels less complex:

```suggestion
use std::os::unix::fs::FileExt;
use std::os::unix::io::FromRawFd;
```